App/Talent.App.WebApp/wwwroot/js/react/ReactScripts/TalentFeed/TalentFeed.jsx

```diff
@@ -11,44 +11,101 @@ export default class TalentFeed extends React.Component {
     };
 
-    //init() {
-    //    let loader = TalentUtil.deepCopy(this.state.loaderData)
-    //    loader.isLoading = false
+    init() {
+        let loaderData = TalentUtil.deepCopy(this.state.loaderData)
+        loaderData.isLoading = false;
+        this.setState({ loaderData });
+        this.loadFeedData();
+    }
 
+    handleScroll() {
+        const win = $(window);
+        if ((($(document).height() - win.height()) == Math.round(win.scrollTop())) || ($(document).height() - win.height()) - Math.round(win.scrollTop()) == 1) {
+            this.setState({ loadingFeedData: true })
+            this.loadFeedData()
+        }
+    };
 
-    //}
+    componentDidMount() {
+        window.addEventListener('scroll', this.handleScroll);
+        this.init()
+    };
 
-    //componentDidMount() {
-    //    window.addEventListener('scroll', this.handleScroll);
-    //    this.init()
-    //};
+    loadFeedData() {
+        const { feedData, loadPosition, loadNumber } = this.state;
+
+        var cookies = Cookies.get('talentAuthToken');
+        $.ajax({
+           
+            url: 'https://mvpstandard-p.azurewebsites.net/profile/profile/getTalent',
+            
+            headers: {
+                'Authorization': 'Bearer ' + cookies,
+                'Content-Type': 'application/json'
+            },
+            data: { position: loadPosition, number: loadNumber },
+            type: "GET",
+            success: function (res) {
+                const joinedData = feedData.concat(res.data);
+                this.setState({
+                    feedData: joinedData,
+                    loadPosition: loadPosition + loadNumber,
+                    loadingFeedData: false,
+                })
+            }.bind(this)
+        })
+    }
 
-   
     render() {
-       
+        const { feedData, loadingFeedData } = this.state;
+
+        return (
+            <BodyWrapper reload={this.init} loaderData={this.state.loaderData}>
+                <div className="ui grid talent-feed container">
+                    <div className="four wide column">
+                        <CompanyProfile />
+                    </div>
+                    <div className="eight wide column ">
+                        {feedData.length > 0 ? feedData.map((feed, index) => (
+                            <TalentCard key={index} feed={feed} />
+                        )) : (
+                            <p>No talent profile to display.</p> 
+                        )}
 
-        //return (
-        //    <BodyWrapper reload={this.init} loaderData={this.state.loaderData}>
-               
-        //    </BodyWrapper>
-        //)
+                        {loadingFeedData ?
+                            <Container fluid className="center aligned">
+                                <Icon loading name='spinner' size="huge" />
+                            </Container>
+                            :
+                            null
+                        }
```

Comment:
As you can see here there are multiple conditions inside render. Instead of doing it like this, you can create new functions.

Reply:
Created new functions renderTalentCards() and renderLoadingSpinner() to call from render function.